src/carts/src/carts-service/dynamo_setup.py

```diff
@@ -0,0 +1,61 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+import os
+import time
+import boto3
+import requests
+import json
+from server import app
+
+
+
+
+# DynamoDB table names passed via environment
+ddb_table_carts = os.getenv("DDB_TABLE_CARTS")
+
+# Allow DDB endpoint to be overridden to support amazon/dynamodb-local
+ddb_endpoint_override = os.getenv("DDB_ENDPOINT_OVERRIDE")
+running_local = False
+
+dynamo_client = None
+
+def verify_local_ddb_running(endpoint):
+    app.logger.info(f"Verifying that local DynamoDB is running at: {endpoint}")
+    for _ in range(5):
+        try:
+            response = requests.get(endpoint)
+            if response.status_code >= 200:
+                app.logger.info("Received HTTP response from local DynamoDB service!")
+                return
+        except requests.ConnectionError:
+            app.logger.info("Local DynamoDB service is not ready yet... pausing before trying again")
+            time.sleep(2)
+    app.logger.error("Local DynamoDB service not responding; verify that your docker-compose .env file is setup correctly")
+    exit(1)
+    
+def setup():
+    global dynamo_client, running_local
+    
+
+    if ddb_endpoint_override:
+        running_local = True
+        app.logger.info("Creating DDB client with endpoint override: " + ddb_endpoint_override)
+        verify_local_ddb_running(ddb_endpoint_override)
+        dynamo_client = boto3.client(
+            'dynamodb',
+            endpoint_url=ddb_endpoint_override,
+            region_name='us-west-2'
```

Comment:
The boto3 client requires a region, either in your aws config environment variables or outlined here like this. In this case since it's the local endpoint it shouldn't really matter what region you put in(it could be any string), but- the local dynamodb has thing where it saves the databases with name, <ACCESS_KEY>_<REGION>. So if you want to query your tables you need to have the right access key and region across your containers. I default to us-west-2 since that is the default region in the env.template and it's easiest to keep track of.